### PR TITLE
2 notification cache factories

### PR DESCRIPTION
**Pull request by andrii-kovalenko-ct**
@lojzatran 
I've added caching to `TenantConfigFactory` and `PaypalPlusFacadeFactory`

If you are agree with the commit and don't have any comments/fix notes - pls, feel free to merge it immediately after your approval

### CODE REVIEW

File: src/test/java/com/commercetools/pspadapter/facade/FactoriesCachingTestIT.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Autowired
+    private PaypalPlusFacadeFactory paypalPlusFacadeFactory;
+
+    /**
+     * Multiple call of tenantConfigFactory.getTenantConfig(config) should return the same instance
+     */
+    @Test
+    public void getTenantConfig_shouldCacheResults() {
+        //
```
Comment (lojzatran) [inline src/test/java/com/commercetools/pspadapter/facade/FactoriesCachingTestIT.java]: remove

Comment (andrii-kovalenko-ct) [reply src/test/java/com/commercetools/pspadapter/facade/FactoriesCachingTestIT.java]: sorry, what to remove?

Comment (lojzatran) [reply src/test/java/com/commercetools/pspadapter/facade/FactoriesCachingTestIT.java]: `//`

Comment (andrii-kovalenko-ct) [reply src/test/java/com/commercetools/pspadapter/facade/FactoriesCachingTestIT.java]: aaaah, sorry, got it e14ad43

**Pull request merged**